openpaas-james/mailbox/encrypted/memory/src/main/java/com/linagora/openpaas/encrypted/InMemoryKeystoreManager.scala

```diff
@@ -0,0 +1,80 @@
+package com.linagora.openpaas.encrypted
+import java.io.ByteArrayInputStream
+
+import com.google.common.io.BaseEncoding
+import com.linagora.openpaas.gpg.Encrypter
+import org.apache.james.core.Username
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import scala.util.Try
+
+class InMemoryKeystoreManager (keystore: scala.collection.concurrent.Map[Username, Set[PublicKey]]) extends KeystoreManager {
+
+  def this() {
+    this(keystore = scala.collection.concurrent.TrieMap())
+  }
+
+  override def save(username: Username, payload: Array[Byte]): Publisher[KeyId] =
+    computeKeyId(payload)
+      .fold(e => SMono.raiseError(new IllegalArgumentException(e)), keyId => usernameExists(username)
+        .fold(_ => create(username, keyId, payload), _ => update(username, keyId, payload)))
+
+  override def listPublicKeys(username: Username): Publisher[PublicKey] =
+    usernameExists(username)
+      .fold(_ => SFlux.empty, _ => SFlux.fromIterable(keystore(username)))
+
+  override def retrieveKey(username: Username, keyId: KeyId): Publisher[PublicKey] =
+    usernameExists(username)
+      .fold(_ => SMono.empty, _ => SMono.fromCallable[Set[PublicKey]](() => keystore(username))
+        .flatMap(keys => find(keyId, keys)))
+
+  override def delete(username: Username, keyId: KeyId): Publisher[Void] =
+    usernameExists(username)
+      .fold(_ => SMono.empty, _ => deleteIfKeyExists(username, keyId))
+
+  override def deleteAll(username: Username): Publisher[Void] =
+    SMono.fromCallable(() => keystore.remove(username)).`then`()
+
+  private def find(keyId: KeyId, keys: Set[PublicKey]): SMono[PublicKey] =
+    keys.find(key => key.id.equals(keyId)).map(SMono.just)
+      .getOrElse(SMono.raiseError(new IllegalArgumentException(s"Cannot find key $keyId")))
+
+  private def deleteIfKeyExists(username: Username, keyId: KeyId): SMono[Void] =
```

Comment:
`deleteIfKeyExists` -> `delete`

The `if` keyword is very unfriendly when we will move to the Cassandra implementation. Let's be less restrictive...